### DISCUSSION TITLE
Dedicated Server Hardware / Performance

### DISCUSSION THREAD

**Discussion by TheDelta** [Aterfax/DCS-World-Dedicated-Server-Docker#77] [Q&A]
Hey,

currently I host a server for my friends so the target audience is relatively small.
While hosting an empty mission with a few enemies it runs fine.
But more complex missions (like through the inferno on syria) run like 🐕💩, to put it nicely.

It's running on debian with the following hardware:

- Intel® Core™ i7-7700 4 cores / 8 threads 3.6 GHz / 4,2 GHz
- 64 GB DDR
- 2 x 4 TB HDD
- Network 1 GBit/s 

There are other services running as well which do take resources, but shutting them temporarily down doesn't make a difference.

After reading some posts, to me the hardware looks fine to host these missions and 3 players.
I already did all optimization steps (removing sam sites, reduce AI amount, etc.) for through the inferno, but performance is still bad.
Lags and delay are insane.

I think about implementing https://github.com/Aterfax/DCS-World-Dedicated-Server-Docker/issues/55 to get some more insight about the situation and understand what the key issue is. 

Could this be related to the hdd?
Are my expectations just too high?
Am I missing something? :D

**Comment by Aterfax** [comment]
Using docker and WINE should not impact the performance hugely (in theory). Docker is "[nearly identical to native performance](https://stackoverflow.com/a/26149994)" (though we might get some additional network latency). I don't question the stated impact of Docker, here.

But the impact of using WINE is somewhat nebulous. Keeping in mind the "WINE is not an emulator" aphorism and that "Wine translates Windows API calls into POSIX calls on-the-fly, eliminating the performance and memory penalties of other methods ." 

The actual performance loss (or gain) when using WINE appears to be answered with: "depends on what you're doing" given a glance around the internet. There's reports that some applications are faster, others are slower, there's the effect of 3D rendering versus 2D rendering versus none at all (as is the case with the DCS Server) etc... etc... etc... 

Ultimately it will be very difficult to tell (or at least get a comprehensive answer) without investigating the performance both in Windows and with Linux using the docker empirically. I've been meaning to put some time aside to investigate this with [DCS-gRPC](https://github.com/DCS-gRPC/rust-server) and [Telemachus](https://github.com/DCS-gRPC/Telemachus) but haven't been able to do so as yet.

If someone else beats me to it that would be excellent.

**Reply by TheDelta** [reply]
yeah that stack looks interesting and maybe this helps me to realize where the issue is with my setup.
I don't understand why it performs so bad right now.

I know the instance is single threaded, but the core speed should be sufficient, yet it's barely playable.

**Comment by Aterfax** [comment]
This may also be good / easier for testing - 

https://gitlab.com/through-the-inferno/dcs-prometheus-exporter
